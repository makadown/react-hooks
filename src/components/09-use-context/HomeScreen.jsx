import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const HomeScreen = () => {

    const miUserContext = useContext(UserContext);

    return (
        <div>
            <h1>Home</h1>
            <hr/>
            { JSON.stringify( miUserContext ) }
        </div>
    )
}
