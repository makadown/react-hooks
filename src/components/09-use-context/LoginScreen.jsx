import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {
    // 1. obtener la referencia al userContext
    const {setUser} = useContext(UserContext);

    const userLogged = {
         name: 'mayito',
         email: 'mayito@mayito.com',
        id: 124
       };
       
    return (
        <div>
            <h1>Login</h1>
            <hr/>
            <button className="btn btn-primary"
                onClick={ () => setUser(userLogged) }>
LOGIN!
            </button>
        </div>
    )
}
