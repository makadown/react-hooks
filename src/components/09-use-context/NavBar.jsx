import React from 'react';
import { NavLink } from 'react-router-dom';

export const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
      <div className="container-fluid">
        <div className="navbar-brand">
          <img src="./logo.png" alt="" width="50px" height="50px" />
        </div>
        <button
          className="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarNav"
          aria-controls="navbarNav"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon"></span>
        </button>
        <div className="collapse navbar-collapse" id="navbarNav">
          <ul className="navbar-nav">
            <li className="nav-item">
              <NavLink exact to="/" className="nav-link">
                Home
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink exact to="./about" className="nav-link">
                About
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink exact to="./login" className="nav-link">
                Login
              </NavLink>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
};
