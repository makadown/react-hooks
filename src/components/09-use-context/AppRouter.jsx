import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from 'react-router-dom';
import { NavBar } from './NavBar';

import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar></NavBar>

        <div className="container mt-3">
          {/* A <Switch> looks through its children <Route>s and
            renders the first one that matches the current URL. */}
          <Switch>
            <Route exact path="/" component={HomeScreen}></Route>
            <Route exact path="/about" component={AboutScreen}></Route>
            <Route exact path="/login" component={LoginScreen}></Route>
            <Route exact path="/home" component={HomeScreen}></Route>
            <Redirect to="/"></Redirect>
          </Switch>
        </div>
      </div>
    </Router>
  );
};
