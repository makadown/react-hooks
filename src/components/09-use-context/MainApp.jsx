import React from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';


export const MainApp = () => {
  const user = {
    id: 1234,
    name: 'Fernando',
    email: 'fhee@gee.com'
  };
  return (
    <UserContext.Provider value={ user }>
      <AppRouter></AppRouter>
    </UserContext.Provider>
  );
};
