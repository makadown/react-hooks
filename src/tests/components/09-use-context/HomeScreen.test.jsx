import React from 'react';
import { mount } from 'enzyme';
import { HomeScreen } from '../../../components/09-use-context/HomeScreen';
import { UserContext } from '../../../components/09-use-context/UserContext';

describe('Pruebas en <HomeScreen/>', () => {
    const user = {
        name: 'mayito_test',
        email: 'mayito@mayito.com'
    };
  const wrapper = mount(
    <UserContext.Provider value={{user}}>
      <HomeScreen />
    </UserContext.Provider>
  );
  test('Debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
