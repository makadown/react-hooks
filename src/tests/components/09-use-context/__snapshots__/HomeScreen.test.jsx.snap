// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen/> Debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Home
  </h1>
  <hr />
  <pre>
    {
   "name": "mayito_test",
   "email": "mayito@mayito.com"
}
  </pre>
</div>
`;
